Mathlib/Algebra/Category/ModuleCat/Sheaf.lean

```diff
@@ -92,6 +92,26 @@ def toSheaf : SheafOfModules.{v} R ⥤ Sheaf J AddCommGrp.{v} where
   obj M := ⟨_, M.isSheaf⟩
   map f := { val := f.val.hom }
 
+/--
+The forgetful functor from sheaves of modules over sheaf of ring `R` to sheaves of `R(X)`-module
+when `X` is initial.
+-/
+@[simps]
+noncomputable def forgetToSheafModuleCat (X : Cᵒᵖ) (hX : Limits.IsInitial X)  :
+    SheafOfModules R ⥤ Sheaf J (ModuleCat (R.1.obj X)) where
+  obj M := ⟨(PresheafOfModules.forgetToPresheafModuleCat X hX).obj M.1, by
+    rw [CategoryTheory.Presheaf.isSheaf_iff_isSheaf_forget J
```

Comment:
Instead of `isSheaf_iff_isSheaf_forget`, I feel it should be possible to apply `isSheaf_iff_isSheaf_comp`(?) to the `forget₂` functor from modules to abelian groups. Then, the goal should be proved `by exact M.isSheaf`.

Reply:
Thanks `isSheaf_iff_isSheaf_comp` indeed works, but I needed to introduce some extra universe constraint. Can you check if this is acceptable please?

Reply:
Could you make explicit the universe `w` in the statement: `SheafOfModules.{w} R ⥤ ...`?

I do not think that your previous version was not more general, because by writing `CategoryTheory.forget.{max v₁ u₁}` in the proof, it was forcing `w := max v₁ u₁`. For the applications to topological spaces (e.g. schemes), this is general enough. (Since I have shown that a limit in `AddCommGrp` exists iff a certain type is small, I think that `forget₂ (ModuleCat _) (AddCommGrp)` reflects all limits, regardless of universes for the index categories. Then, `forgetToSheafModuleCat` should be defined regardless of the universes, but I do not feel it is necessary to do it now!)

Reply:
In #14283, I show that the forget functor from `ModuleCat` to `AddCommGrp` reflects all limits (regardless of universes) [for the preservation we may need that the ring is `v`-small]. Then, I think that you may now use `isSheaf_of_isSheaf_comp` applied to this `forget₂` in order to remove universe restrictions.

Reply:
Thanks! This works perfectly!